.github/scripts/generate_ci_workflows.py

```diff
@@ -169,6 +180,7 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
         cuda_version="10.1",
         test_runner_type=WINDOWS_CUDA_TEST_RUNNER,
         on_pull_request=True,
+        only_run_smoke_tests_on_pull_request=True,
         num_test_shards=2,
```

Comment:
wouldn't it be simply to juse make num_test_shards=1 here and a comment saying this only runs smoke_tests? (this way we dont need to have complex logic to figure out num_test_shards.

Reply:
No, because on master we'd still want 2 shards.